app/controllers/miq_ae_tools_controller.rb

```diff
@@ -196,57 +196,70 @@ def review_import
   end
 
   def retrieve_git_datastore
-    redirect_options = {:action => :review_git_import}
     git_url = params[:git_url]
-    verify_ssl = params[:git_verify_ssl] == "true" ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE
-    new_git_repo = false
 
     if git_url.blank?
       add_flash(_("Please provide a valid git URL"), :error)
+      response_json = {:message => @flash_array.first}
     elsif !GitBasedDomainImportService.available?
       add_flash(_("Please enable the git owner role in order to import git repositories"), :error)
+      response_json = {:message => @flash_array.first}
     else
       begin
-        git_repo = GitRepository.find_or_create_by!(:url => git_url) { new_git_repo = true }
-        git_repo.update_attributes(:verify_ssl => verify_ssl)
-        if params[:git_username] && params[:git_password]
-          git_repo.update_authentication(:values => {:userid   => params[:git_username],
-                                                     :password => params[:git_password]})
-        end
+        setup_results = git_repository_service.setup(
+          git_url,
+          params[:git_username],
+          params[:git_password],
+          params[:git_verify_ssl]
+        )
+        git_repo_id = setup_results[:git_repo_id]
+        new_git_repo = setup_results[:new_git_repo?]
 
-        task_options = {
-          :action => "Retrieve git repository",
-          :userid => current_user.userid
-        }
-        queue_options = {
-          :class_name  => "GitRepository",
-          :method_name => "refresh",
-          :instance_id => git_repo.id,
-          :role        => "git_owner",
-          :args        => []
-        }
-
-        task_id = MiqTask.generic_action_with_callback(task_options, queue_options)
-        task = MiqTask.wait_for_taskid(task_id)
-
-        raise task.message unless task.status == "Ok"
-
-        branch_names = git_repo.git_branches.collect(&:name)
-        tag_names = git_repo.git_tags.collect(&:name)
-        redirect_options[:git_branches] = branch_names.to_json
-        redirect_options[:git_tags] = tag_names.to_json
-        redirect_options[:git_repo_id] = git_repo.id
-        flash_message = "Successfully found git repository, please choose a branch or tag"
-        add_flash(_(flash_message), :success)
+        task_id = git_based_domain_import_service.queue_refresh(git_repo_id)
+        response_json = {:task_id => task_id, :git_repo_id => git_repo_id, :new_git_repo => new_git_repo}
       rescue => err
-        git_repo.destroy if git_repo && new_git_repo
```

Comment:
@mkanoor It's being taken care of in the check_git_task now: https://github.com/ManageIQ/manageiq/pull/12964/files/dbff1b629627610e45ed00d9f7bc9728ba6205e4#diff-6a59cc1ea26326d3a83ec0fddaf40487R251